MapleServer2/Types/Currency.cs

```diff
@@ -0,0 +1,70 @@
+﻿using MapleServer2.Enums;
+using MapleServer2.Packets;
+
+namespace MapleServer2.Types
+{
+    public class Currency
+    {
+        private readonly Player Player;
+        private readonly CurrencyType Type;
+        public long Amount { get; private set; }
+
+        public Currency(Player player, CurrencyType type, long input)
+        {
+            Player = player;
+            Type = type;
+            Amount = input;
+        }
+
+        public bool Modify(long input)
+        {
+            if (Amount + input < 0)
+            {
+                return false;
+            }
+            Amount += input;
+            UpdateUi();
+            return true;
+        }
+
+        public void SetAmount(long input)
+        {
+            if (input < 0)
+            {
+                return;
+            }
+            Amount = input;
+            UpdateUi();
+        }
+
+        private void UpdateUi()
+        {
+            switch (Type)
+            {
+                case CurrencyType.Meso:
+                    Player.Session.Send(MesosPacket.UpdateMesos(Player.Session));
+                    break;
+                case CurrencyType.Meret:
+                case CurrencyType.GameMeret:
+                case CurrencyType.EventMeret:
+                    Player.Session.Send(MeretsPacket.UpdateMerets(Player.Session));
+                    break;
+                case CurrencyType.ValorToken:
+                case CurrencyType.Treva:
+                case CurrencyType.Rue:
+                case CurrencyType.HaviFruit:
+                case CurrencyType.MesoToken:
+                case CurrencyType.BlueStar:
+                case CurrencyType.RedStar:
+                case CurrencyType.EventDungeonCoin:
+                case CurrencyType.GuildCoins:
+                case CurrencyType.KayCoin:
+                case CurrencyType.MapleCoin:
+                case CurrencyType.PremiumCoin:
```

Comment:
Any coin that is an item is not considered as a currency by the server, it is just an item so these should be removed.

Reply:
Do you know which ones are items?

Reply:
All of the ones I put in the comment.

Reply:
iirc, the only non-item currencies are: mesos, merets, valor, treva, rue, havi fruit, and meso tokens.